File: src/worker/java/MyWorker.java
import parspice.sender.DoubleArraySender;
import parspice.worker.OWorker;

public class MyWorker extends OWorker<double[]> {

    public MyWorker() {
        super(new DoubleArraySender(3));
    }

    @Override
    public void setup() throws Exception {

    }

    @Override
    public double[] task(int i) throws Exception {
        return new double[]{1, 2, i};
    }
}

File: src/worker/kotlin/MyWorker.kt
import parspice.worker.OWorker
import parspice.sender.DoubleArraySender

class MyWorker: OWorker<DoubleArray>(DoubleArraySender(3)) {
    override fun setup() {}

    override fun task(i: Int) = doubleArrayOf(1.0, 2.0, i.toDouble())
}